code/modules/projectiles/guns/ballistic.dm

```diff
@@ -44,6 +44,19 @@
 			allowed_magazines += /obj/item/ammo_magazine/smart
 	update_icon()
 
+/obj/item/gun/ballistic/update_icon_state()
+	. = ..()
+	if(magazine_type && ammo_magazine)
+		if(silenced)
+			icon_state = "[silenced_icon]"
+		else
+			icon_state = "[initial(icon_state)]"
+	else if(magazine_type && !ammo_magazine)
+		if(silenced)
+			icon_state = "[silenced_icon]-empty"
+		else
+			icon_state = "[initial(icon_state)]-empty"
+
```

Comment:
I feel like this entire conditional section could be written a bit cleaner in a form something like this:

```
if(magazine_type)
    var/silenced_state = silenced ? silenced_icon : initial(icon_state)
	var/magazine_state = ammo_magazine ? "-empty" : ""
	icon_state = "[silenced_state][magazine_state]"
```